java/deepgramTopicDetection/Index.java

```diff
@@ -0,0 +1,171 @@
+import java.util.Collections;
+import java.util.stream.Collectors;
+import java.util.*;
+import java.util.stream.*;
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+
+import com.google.gson.Gson;
+import org.apache.commons.validator.routines.UrlValidator;
+final Gson gson=new Gson();
+
+public RuntimeResponse main(RuntimeRequest req,RuntimeResponse res)throws Exception{
+        // Validate that values present in the request are not empty (payload, variables)
+        RuntimeResponse errorResponse=checkEmptyPayloadAndVariables(req,res);
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        // Validate the requested payload (URL)
+        String payloadString=req.getPayload().toString();
+        Map<String, Object> payload=gson.fromJson(payloadString,Map.class);
+
+        errorResponse=validatePayload(payload,res);
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        // Get file url from payload
+        String fileurl=payload.get("fileUrl").toString();
+
+        // Name API key is not empty
+        String apiKeyVariable="DEEPGRAM_SECRET_KEY";
+
+        errorResponse=checkEmptyAPIKey(req,res,apiKeyVariable);
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        String apiKey=req.getVariables().get(apiKeyVariable).toString();
+
+        String deepgramData="";
+        Map<String, Object> responseData=new HashMap<>();
+
+        try{
+            deepgramData=generateSummary(fileurl,apiKey);
+        }catch(Exception e){
+            responseData.put("success",false);
+            responseData.put("message","Something went wrong while generating the summary, please check with the developers. Error: "+e.getMessage());//TODO
+        return res.json(responseData);
+        }
+        responseData.put("success",true);
+        responseData.put("deepgramData",deepgramData);
```

Comment:
Looks like this is being returned as a string. Would you please return it as a map instead?